cmd/qq/main.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/hpcloud/tail"
+)
+
+var (
+	logfile string
+	logpath string
+)
+
+func init() {
+	logfile = "q"
+	setPath()
+}
+
+func main() {
+	tailFile()
+}
+
+func tailFile() {
+	t, err := tail.TailFile(logpath, tail.Config{Follow: true})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for line := range t.Lines {
+		fmt.Println(line.Text)
+	}
+}
+
+func setPath() {
+	tmpdir := os.Getenv("TMPDIR")
+	if tmpdir == "" {
+		if _, err := os.Stat("/system/bin/adb"); os.IsNotExist(err) {
+			// Handle android
+			logpath = "/data/local/tmp/" + logfile
+		} else {
+			logpath = "/tmp/" + logfile
+		}
+	} else {
+		logpath = tmpdir + logfile
+	}
+}
```

Comment:
`setPath` can be replaced with a call to [os.TempDir()](https://golang.org/src/os/file_unix.go#L336), which does exactly this.

Also, instead of assigning to global vars, you should just have `setPath` return a `string`.